packages/utils/src/dal/mikro-orm/mikro-orm-create-connection.ts

```diff
@@ -1,28 +1,39 @@
 import { ModuleServiceInitializeOptions } from "@medusajs/types"
 
 export async function mikroOrmCreateConnection(
-  database: ModuleServiceInitializeOptions["database"],
+  database: ModuleServiceInitializeOptions["database"] & { connection?: any },
   entities: any[]
 ) {
-  const { MikroORM } = await import("@mikro-orm/postgresql")
+  let schema = database.schema || "public"
+
+  let driverOptions = database.driverOptions ?? {
+    connection: { ssl: true },
+  }
+
+  let clientUrl = database.clientUrl
+
+  if (database.connection) {
+    // Reuse already existing connection
+    // It is important that the knex package version is the same as the one used by MikroORM knex package
```

Comment:
would it make sense to use this instead then? https://www.npmjs.com/package/@mikro-orm/knex

Reply:
No cause it comes from another place and not necessarily a place that is aware of or use mikro orm. Plus, the mikro orm knex package encapsulates other things as well that we don't really care of here, since we just want to create a connection. Also, in the mikro orm package they check the instanceof the knex.Client, so lets just download what we need wdyt? I would also understand the purpose if it can prevent us from having to take care of the version ourselves 😅

Reply:
Fair enough. In this magical place another, we do however need to be aware of mikroorm though, since it the knex package needs to be the same, just not package-wise. But even in the mikroorm package, we need to be version aware, so that doesn't really fix the issue. Will resolve this

Reply:
Interesting. Might be that we're using the global manager somewhere, which should be a bug. We can push this for later.